UPGRADING.md

```diff
@@ -36,6 +37,100 @@ The following naming convention has been adopted for Elasticsearch.
+
+   }
+   ```
+
+2. Add `Setting.Property.Deprecated` to each existing setting to mark them deprecated. For example:
+
+   ```java
+   public class LegacyOpenDistroJobSchedulerSettings {
+
+      public static final Setting<TimeValue> REQUEST_TIMEOUT = Setting.positiveTimeSetting(
+               "opendistro.jobscheduler.request_timeout",
+               TimeValue.timeValueSeconds(10),
+               Setting.Property.NodeScope, Setting.Property.Dynamic, Setting.Property.Deprecated);
+
+   }
+   ```
+
+3. Include legacy settings in `Plugin#getSettings`. Example:
+
+   ```java
+   public class JobSchedulerPlugin extends Plugin implements ExtensiblePlugin {
+
+      @Override
+      public List<Setting<?>> getSettings() {
+         List<Setting<?>> settingList = new ArrayList<>();
+         settingList.add(LegacyOpenDistroJobSchedulerSettings.REQUEST_TIMEOUT);
+         settingList.add(JobSchedulerSettings.REQUEST_TIMEOUT);
+         return settingList;
+      }
+   }
+   ```
+
+   If you do not add your legacy settings here they will be archived upon the node joining the OpenSearch cluster.
+   
+4. In the non-legacy class, rename settings and re-declare them to fallback on the legacy settings.
+
+   ```java
+   public class JobSchedulerSettings {
+
+       public static final Setting<TimeValue> REQUEST_TIMEOUT = Setting.positiveTimeSetting(
+            "opensearch.jobscheduler.request_timeout",
+            LegacyOpenDistroJobSchedulerSettings.REQUEST_TIMEOUT,
+            Setting.Property.NodeScope, Setting.Property.Dynamic);
+
+   }
+   ```
+
+   Note that this setting does not have a default value, but defaults to falling back to `LegacyOpenDistroJobSchedulerSettings.REQUEST_TIMEOUT`.
+
+5. Make sure you have implemented a listener to update the setting value. If a setting is set in `opensearch.yml`, you're all set. But if the setting is set via API, the node will start, settings will be read, defaults applied (fallback value will be used), and only afterwards fetched from the cluster and set on your node. Thus your setting will end up with the default value of the fallback.
+
+   To fix this, add a listener. For example:
+
+   ```java
+   clusterService.getClusterSettings().addSettingsUpdateConsumer(JobSchedulerSettings.REQUEST_TIMEOUT,
+      value -> {
+         this.requestTimeout = value;
+         log.debug("Setting request timeout: " + this.requestTimeout.getMinutes());
+      });
+   ```
+
+6. Write tests.
+
+   ```java
+    public void testSettingsGetValueWithLegacyFallback() {
+        Settings settings = Settings.builder()
+            .put("opendistro.jobscheduler.retry_count", 3)
+        .build();
+        
+        assertEquals(JobSchedulerSettings.RETRY_COUNT.get(settings), Integer.valueOf(3)); 
+
+        assertSettingDeprecationsAndWarnings(new Setting[]{
+            LegacyOpenDistroJobSchedulerSettings.RETRY_COUNT
+        });
+    }   
+   ```
+
+7. Document your newly renamed settings.
+
+   TODO
```

Comment:
@aetter what should we write here? Where do I make changes for job-scheduler settings public documentation?

Reply:
Hey @dblock, we actually have no user-facing content on job-scheduler since it plays such a supporting role. We've deprioritized it early on and have never had a reader ask for anything. I'm happy to add a new top-level section to the documentation if you have content that you'd like to get out there. Just let me know. Otherwise, the most likely place is the job-scheduler README.

Reply:
@aetter  What's another plugin link/example that we can put here?

Reply:
Alerting is a decent example since the settings page also includes the managed indices and their purposes, which I assume might change during a migration. https://docs-beta.opensearch.org/docs/alerting/settings/#alerting-settings

Reply:
We have a link in the lower left to "edit this page" that should send you to the source. In this case, it's https://github.com/opensearch-project/documentation-website/blob/main/docs/alerting/settings.md